src/configure/test/suite/GitHubClient.test.ts

```diff
@@ -0,0 +1,135 @@
+import { GithubClient } from "../../clients/github/githubClient";
+
+var expect = require('chai').expect;
+var nock = require('nock');
+
+let org = "Organization";
+let repoName = "Repository";
+let githubClient = new GithubClient("PAT", "repoUrl");
+
+
+describe('# Testing listOrganizations() ', function () {
+    this.timeout(500000);
+
+    before(function () {
+        var orgList = [
+            {
+                "login": "SampleOrganizationA",
+                "id": 66720950,
+                "url": "https://api.github.com/orgs/SampleOrganizationA",
+                "repos_url": "https://api.github.com/orgs/SampleOrganizationA/repos"
+            },
+            {
+                "login": "SampleOrganizationB",
+                "id": 67440137,
+                "url": "https://api.github.com/orgs/SampleOrganizationB",
+                "repos_url": "https://api.github.com/orgs/SampleOrganizationB/repos",
+            },
+            {
+                "login": "SampleOrganizationC",
+                "id": 67453753,
+                "url": "https://api.github.com/orgs/SampleOrganizationC",
+                "repos_url": "https://api.github.com/orgs/SampleOrganizationC/repos"
+            }];
+
+        nock('https://api.github.com')
+            .get('/user/orgs')
+            .reply(200, orgList);
+    });
+
+    context('User is a member of organizations(s)', function () {
+        it('should return a list of organizations', function (done) {
+            githubClient.listOrganizations(true).then((orgs) => {
+                expect(Array.isArray(orgs)).to.equal(true);
+                expect(orgs.length).to.equal(3);
+                orgs.forEach((org) => {
+                    expect(org).to.have.property('login');
+                    expect(org).to.have.property('id');
+                    expect(org).to.have.property('url');
+                });
+                done();
+            });
+        });
+    });
+    before(function () {
```

Comment:
can we have multiple `before` in one `describe`? I think we should not do that as this could be confusing and could lead to errors. I see three options here:
1. Create a separate `describe` block
2. remove this second `before` and move the `nock` statement to inside the test itself
3. add a function called getOrgList() which will either return valid list or an empty list based on an environment variable. before will use this function inside nock call. each test can set proper environment variables to get desired response.

Reply:
I prefer #3. But this requires that nock support calling a function to get mocked response

Reply:
Option 1 - creating a separate describe block for each test seems like the best option because the `nock` interceptor needs to be defined inside `before` block due to which 2nd option is not feasible, and 3rd option might not provide the clarity of code which we are trying to target.

Reply:
But the main purpose behind putting multiple tests and hence multiple `before`s inside single `describe` was to create a set of tests that test different cases of a single function. Should I change this approach?